bundles/org.openhab.binding.mqtt.homie/src/main/java/org/openhab/binding/mqtt/homie/internal/homie300/Property.java

```diff
@@ -187,13 +188,16 @@ public void createChannelFromAttribute() {
                 BigDecimal min = s.length == 2 ? convertFromString(s[0]) : null;
                 BigDecimal max = s.length == 2 ? convertFromString(s[1]) : null;
                 BigDecimal step = (min != null && max != null)
-                        ? max.subtract(min).divide(new BigDecimal(100.0), new MathContext(isDecimal ? 2 : 0))
-                        : null;
+                           ? max.subtract(min).divide(new BigDecimal(100.0), new MathContext(isDecimal ? 2 : 0))
+                           : null;
                 if (step != null && !isDecimal && step.intValue() <= 0) {
-                    step = new BigDecimal(1);
+                        step = new BigDecimal(1);
+                }
+                if(attributes.unit != null && attributes.format.contains("%")) {
```

Comment:
Why are you checking if the 'format' field contains the "%" sign? Shouldn't this be:

```java
if (attributes.unit != null && attributes.unit.contains("%")) {
```

Homie V4 specification defines that, for integer and float, "format" can only describe 'a range of payloads e.g. 10:15' (https://homieiot.github.io/specification/#property-attributes).

Perhaps there is something I'm misssing :thinking:

Reply:
I forgot to add the latest changes to this branch. I will push them as I changed this already to unit.

Not really, when I tested it OpenHab calculated the value from the format range from the Percentage value. So if your format is 0:200 and you have the Dimmer set to 50% the value which will be sent to MQTT is 100. I find it useful for values where you need dimmmer functionality but your range goes from 0 to 255 and OpenHab should display it as Dimmer.